src/auth/auth.service.ts

```diff
@@ -0,0 +1,57 @@
+import { BadRequestException, ForbiddenException, Injectable, UnauthorizedException } from "@nestjs/common";
+import { JwtService } from "@nestjs/jwt";
+import { MockUserService } from "../mockModules/mock-user/mock-user.service";
+import { UserRolesEnum } from "@prisma/client";
+import * as bcrypt from "bcrypt";
+import { CreateAuthDto } from "./dto/create-auth.dto";
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private jwtService: JwtService,
+    private userService: MockUserService
+  ) {}
+
+  public async signIn(createAuthDto:CreateAuthDto, role: UserRolesEnum) {
+    // Check email is valid or not
+    const user = await this.userService.findOneByEmail(createAuthDto.email);
+    // If not a valid user then thow an error
+    if (!user) {
+      // Handle the case when the user is not found.
+      throw new Error("User not found");
+    }
+    // Compare the entered password with the password fetched from database
+    const isPasswordValid = await this.comparePasswords(
+      user.password,
+      createAuthDto.password
+    );
+    // Check password is valid or not
+    if (!isPasswordValid) throw new BadRequestException("Incorrect password");
```

Comment:
Throw UnauthorizedException and "Invalid credential" as the error message

Reply:
Done